dipy/align/tests/test_transforms.py

```diff
@@ -0,0 +1,233 @@
+    sb = np.sin(theta[1])
+    cc = np.cos(theta[2])
+    sc = np.sin(theta[2])
+
+    X = np.array([[1,  0,  0 ],
+                  [0, ca, -sa],
+                  [0, sa,  ca]])
+    Y = np.array([[cb,  0, sb],
+                  [0,   1,  0],
+                  [-sb, 0, cb]])
+    Z = np.array([[cc, -sc, 0],
+                  [sc,  cc, 0],
+                  [0 ,   0, 1]])
+
+    R = Z.dot(X.dot(Y)) # Apply in order: Y, X, Z (Y goes to the right)
+    expected = np.eye(dim + 1)
+    expected[:3, :3] = R[:3, :3]
+    actual = np.empty_like(expected)
+    param_to_matrix(ttype, dim, theta, actual)
+    assert_array_almost_equal(actual, expected)
+
+    # Test rigid matrix 3D
+    dim = 3
+    ttype = transform_type['RIGID']
+    theta = np.random.rand(6)
+    ca = np.cos(theta[0])
+    sa = np.sin(theta[0])
+    cb = np.cos(theta[1])
+    sb = np.sin(theta[1])
+    cc = np.cos(theta[2])
+    sc = np.sin(theta[2])
+
+    X = np.array([[1,  0,  0 ],
+                  [0, ca, -sa],
+                  [0, sa,  ca]])
+    Y = np.array([[cb,  0, sb],
+                  [0,   1,  0],
+                  [-sb, 0, cb]])
+    Z = np.array([[cc, -sc, 0],
+                  [sc,  cc, 0],
+                  [0 ,   0, 1]])
+
+    R = Z.dot(X.dot(Y)) # Apply in order: Y, X, Z (Y goes to the right)
+    expected = np.eye(dim + 1)
+    expected[:3, :3] = R[:3, :3]
+    expected[:3, 3] = theta[3:6]
+    actual = np.empty_like(expected)
+    param_to_matrix(ttype, dim, theta, actual)
+    assert_array_almost_equal(actual, expected)
+
+    # Test scaling matrix 2D
+    dim = 3
+    ttype = transform_type['SCALING']
+    factor = np.random.rand()
+    theta = np.array([factor])
+    expected = np.array([[factor, 0, 0, 0],
+                         [0, factor, 0, 0],
+                         [0, 0, factor, 0],
+                         [0, 0, 0, 1]])
+    actual = np.empty_like(expected)
+    param_to_matrix(ttype, dim, theta, actual)
+    assert_array_almost_equal(actual, expected)
+
+    # Test affine 2D
+    dim = 3
+    ttype = transform_type['AFFINE']
+    theta = np.random.rand(12)
+    expected = np.eye(4)
+    expected[0,:] = theta[:4]
+    expected[1,:] = theta[4:8]
+    expected[2,:] = theta[8:12]
+    actual = np.empty_like(expected)
+    param_to_matrix(ttype, dim, theta, actual)
+    assert_array_almost_equal(actual, expected)
+
+
+def test_get_identity_parameters():
+    transforms = transform_type.keys()
+    for dim in [2, 3]:
+        for transform in transforms:
+            t = transform_type[transform]
+            n = number_of_parameters(t, dim)
+            theta = np.empty(n)
+            get_identity_parameters(t, dim, theta)
+
+            expected = np.eye(dim + 1)
+            actual = np.empty_like(expected)
+            param_to_matrix(t, dim, theta, actual)
+            assert_array_almost_equal(actual, expected)
+
+
+def test_eval_jacobian_function():
+    r""" Compare the analytical Jacobians with their numerical approximations
```

Comment:
Comments better than docstrings at the top, because the docstring gets printed out during nostests, and this make it harder to see which function is being tested, and doesn't match the output for functions without docstrings.